### PR TITLE
Allow administrative monitors to be displayed for users with `Overall/MANAGE` permission

### PR DESCRIPTION
**Pull request by Vlatombe**
This allows users with `Overall/MANAGE` permission but not `Overall/SYSTEM_READ` to display administrative monitors if the administrative monitor has been implemented to allow it.

### Testing done

Verified with a custom administrative monitor allowing `Overall/MANAGE` that it can be displayed in the corresponding UI area by a user with this permission.

### Proposed changelog entries

- Allow some administrative monitors to be displayed for users with `Overall/MANAGE` permission

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [X] There is automated testing or an explanation as to why this change has no tests.
- [X] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/hudson/model/AdministrativeMonitor.java [hunk 1]
```diff
@@ -183,7 +183,7 @@ public void doDisable(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
 
     /**
      * Required permission to view this admin monitor.
-     * By default {@link Jenkins#ADMINISTER}, but {@link Jenkins#SYSTEM_READ} is also supported.
+     * By default {@link Jenkins#ADMINISTER}, but {@link Jenkins#SYSTEM_READ} or {@link Jenkins#MANAGE} are also supported.
```
Comment (jglick) [inline core/src/main/java/hudson/model/AdministrativeMonitor.java:186]: But you cannot declare a monitor to appear for users with _either_ `SYSTEM_READ` _or_ `MANAGE` (suppressing any fix GUI in the former case). This seems like a significant limitation, though better than requiring `ADMINISTER`.

File: test/src/test/java/jenkins/management/AdministrativeMonitorsDecoratorTest.java [hunk 2]
```diff
@@ -43,6 +51,16 @@ public class AdministrativeMonitorsDecoratorTest {
     @Rule
     public JenkinsRule j = new JenkinsRule();
 
+    @BeforeClass
+    public static void enablePermissions() {
+        System.setProperty("jenkins.security.ManagePermission", "true");
```
Comment (jglick) [inline test/src/test/java/jenkins/management/AdministrativeMonitorsDecoratorTest.java:56]: `FlagRule`?

File: core/src/main/java/hudson/model/AdministrativeMonitor.java [hunk 3]
```diff
@@ -201,13 +205,36 @@ public Permission getRequiredPermission() {
         return Jenkins.ADMINISTER;
```
Comment (Vlatombe) [inline core/src/main/java/hudson/model/AdministrativeMonitor.java:205]: Hesitating to deprecate this method, but it still matches simple use cases so went for a javadoc instead.

File: core/src/main/java/hudson/model/AdministrativeMonitor.java (hunk 1 above)
Comment (Vlatombe) [reply core/src/main/java/hudson/model/AdministrativeMonitor.java:186]: Implemented, but requires 2 separate methods.

File: core/src/main/java/hudson/model/AdministrativeMonitor.java [hunk 4]
```diff
@@ -183,14 +183,18 @@ public void doDisable(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
...
+     * By default {@link Jenkins#ADMINISTER}, but {@link Jenkins#SYSTEM_READ} or {@link Jenkins#MANAGE} are also supported.
      * <p>
      *     Changing this permission check to return {@link Jenkins#SYSTEM_READ} will make the active
      *     administrative monitor appear on {@code manage.jelly} and on the globally visible
      *     {@link jenkins.management.AdministrativeMonitorsDecorator} to users without Administer permission.
      *     {@link #doDisable(StaplerRequest, StaplerResponse)} will still always require Administer permission.
      * </p>
      * <p>
+     *     This method only allows for a single permission to be returned. If more complex permission checks are required,
+     *     override {@link #checkRequiredPermission()} and {@link #hasRequiredPermission()} instead.
```
Comment (jglick) [inline core/src/main/java/hudson/model/AdministrativeMonitor.java:195]: Note that these methods are very similar to `AccessControlled`, but creating a custom `ACL` seems more laborious than implementing them as written.

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
What about `AdministrativeMonitorsConfiguration` and https://github.com/jenkinsci/jenkins/blob/dc36857468f76e74ea7686bd863b2943d0107223/core/src/main/java/hudson/model/AdministrativeMonitor.java#L178-L179? With this change, IIUC, users with only `MANAGE` could see monitors and agree to fix the issue the monitor warns about, but they could not dismiss monitors they found irrelevant.

**Review by amuniz — APPROVED** [review]

File: core/src/main/java/hudson/model/AdministrativeMonitor.java [hunk 5]
```diff
@@ -201,13 +205,50 @@ public Permission getRequiredPermission() {
         return Jenkins.ADMINISTER;
     }
 
+    /**
+     * Checks if the current user has the minimum required permission to view this administrative monitor.
+     * <p>
+     * Subclasses may override this method and {@link #hasRequiredPermission()} instead of {@link #getRequiredPermission()} to perform more complex permission checks,
+     * for example, checking either {@link Jenkins#MANAGE} or {@link Jenkins#SYSTEM_READ}.
```
Comment (jglick) [inline core/src/main/java/hudson/model/AdministrativeMonitor.java:212]: The Javadoc is misleading as it implies that you could grant permission to see admin monitors based on some other criteria. But in fact due to `hasPermissionToDisplay` these are really the only permissions you can check.

**Pull request merged**